sdks/python/apache_beam/coders/coders.py

```diff
@@ -280,10 +280,18 @@ def from_runner_api(cls, coder_proto, context):
     Prefer registering a urn with its parameter type and constructor.
     """
     parameter_type, constructor = cls._known_urns[coder_proto.spec.spec.urn]
-    return constructor(
-        proto_utils.parse_Bytes(coder_proto.spec.spec.payload, parameter_type),
-        [context.coders.get_by_id(c) for c in coder_proto.component_coder_ids],
-        context)
+    try:
+      return constructor(
+          proto_utils.parse_Bytes(
+              coder_proto.spec.spec.payload, parameter_type),
+          [context.coders.get_by_id(c)
+           for c in coder_proto.component_coder_ids],
+          context)
+    except:
```

Comment:
Bare excepts are bad, and can catch things like control-C, etc. 

Also, if the URN was in known_urns, we should expect the constructor to succeed. Only when it's not is it "external" and we should fall back to the Holder.